### PR TITLE
adding Cov Matrices to track propagation

### PR DESCRIPTION
**Pull request by nzardosh**
adding covariant matrix to track propagation task

adding track covariance propagation

### CODE REVIEW

**Comment by jgrosseo** [comment]
Can you fix clang?

File: Common/TableProducer/trackPropagation.cxx [hunk 1]
```diff
@@ -62,9 +62,13 @@ struct TrackPropagation {
   bool fillTracksParPropagated = false;
   bool fillTracksExtended = false;
 
+  int runNumber;
```
Comment (jgrosseo) [inline Common/TableProducer/trackPropagation.cxx:65]: Please remove, including the comment code below

File: Common/TableProducer/trackPropagation.cxx [hunk 2]
```diff
@@ -62,9 +62,13 @@ struct TrackPropagation {
...
+
   o2::base::Propagator::MatCorrType matCorr;
 
   const o2::dataformats::MeanVertexObject* mVtx;
+  o2::parameters::GRPObject* grpo;
+  o2::base::MatLayerCylSet* lut;
```
Comment (jgrosseo) [inline Common/TableProducer/trackPropagation.cxx:71]:
```suggestion
  o2::parameters::GRPObject* grpo = nullptr;
  o2::base::MatLayerCylSet* lut = nullptr;
```

**Comment by nzardosh** [comment]
> There is no table for the CovMatrix yet, right? (This is ok for now, just to confirm)

For the matrix itself no, but the tracks can be propagated using the covariance info

**Review by jgrosseo — APPROVED** [review]

**Pull request merged**